Settings.c

```diff
@@ -73,20 +73,27 @@ static void writeQuotedList(FILE* fp, char** list) {
 
 */
 
-void Settings_delete(Settings* this) {
-   free(this->filename);
-   free(this->initialFilename);
-   for (unsigned int i = 0; i < HeaderLayout_getColumns(this->hLayout); i++) {
+static void Settings_deleteColumns(Settings* this) {
+   for (size_t i = 0; i < HeaderLayout_getColumns(this->hLayout); i++) {
       String_freeArray(this->hColumns[i].names);
       free(this->hColumns[i].modes);
    }
    free(this->hColumns);
+}
```

Comment:
This should set `this->hColumns = NULL;` explicitly, to avoid potential UAF issues.

Reply:
That is not a pattern followed by the rest of the code base (in every _delete routine I just quickly audited at least), and I'm not particularly keen to start introducing it here TBH.  Better to use tools like valgrind IMO.